src/client/runner/automation/playback/click/select-child-click.js

```diff
@@ -0,0 +1,203 @@
+import hammerhead from '../../../deps/hammerhead';
+import testCafeCore from '../../../deps/testcafe-core';
+import testCafeUI from '../../../deps/testcafe-ui';
+import * as automationSettings from '../../settings';
+import MoveAutomation from '../move';
+import MoveOptions from '../../options/move';
+import delay from '../../../utils/delay';
+import * as mouseUtils from '../../../utils/mouse';
+
+var Promise = hammerhead.Promise;
+
+var browserUtils     = hammerhead.utils.browser;
+var eventSimulator   = hammerhead.eventSandbox.eventSimulator;
+var focusBlurSandbox = hammerhead.eventSandbox.focusBlur;
+
+var domUtils   = testCafeCore.domUtils;
+var styleUtils = testCafeCore.styleUtils;
+
+var selectElementUI = testCafeUI.selectElement;
+
+
+const FOCUS_DELAY = browserUtils.hasTouchEvents ? 0 : 160;
+
+export default class SelectChildClickAutomation {
+    constructor (element, clickOptions) {
+        this.element   = element;
+        this.modifiers = clickOptions.modifiers;
+        this.caretPos  = clickOptions.caretPos;
+
+        this.offsetX = clickOptions.offsetX;
+        this.offsetY = clickOptions.offsetY;
+
+        this.parentSelect       = domUtils.getSelectParent(this.element);
+        this.optionListExpanded = this.parentSelect ? selectElementUI.isOptionListExpanded(this.parentSelect) : false;
+        this.childIndex         = null;
+
+        if (this.parentSelect) {
+            this.childIndex = this.element.tagName.toLowerCase() === 'option' ?
+                              domUtils.getOptionIndex(this.parentSelect, this.element) :
+                              domUtils.getOptionGroupIndex(this.parentSelect, this.element);
+        }
+
+        var selectedIndex = this.parentSelect.selectedIndex;
+        var isOption      = this.element.tagName.toLowerCase() === 'option';
```

Comment:
we can define this above to avoid `this.element.tagName.toLowerCase() === 'option'` duplication